src/main/java/org/fsg1/fmms/backend/endpoints/CurriculumEndpoint.java

```diff
@@ -45,6 +45,25 @@ public Response getCurriculumSemesters(@PathParam("curriculum_id") final int cur
     }
 
     /**
+     * Get the complete information of one semester in a curriculum.
+     *
+     * @param curriculumId Identifier of the curriculum.
+     * @param semesterId   Identifier of the semester.
+     * @throws Exception In case the querying goes wrong.
+     * @return A JSON Object containing the complete information of this semester.
+     */
+    @GET
+    @Path("/semesters/{semester_id}")
```

Comment:
If you are referring to one semester by providing the semester id or number, please use singular.

Reply:
This is defined by @sjoerdbr . 

Also the .../modules/module_id endpoint uses plural too.

Reply:
Forget it